server/punishments.ts

```diff
@@ -772,11 +774,17 @@ export const Punishments = new class {
 		const userid = toID(user);
 		if (Users.get(user)?.locked) return false;
 		const name = typeof user === 'string' ? user : (user as User).name;
+		let outputPunishment;
+		let affected;
 		if (namelock) {
 			punishment = `NAME${punishment}`;
-			await Punishments.namelock(user, expires, toID(namelock), false, `Autonamelock: ${name}: ${reason}`);
+			({
+				punishment: outputPunishment, affected,
+			} = await Punishments.namelock(user, expires, toID(namelock), false, `Autonamelock: ${name}: ${reason}`));
```

Comment:
```suggestion
			{
				punishment: outputPunishment,
				affected,
			} = await Punishments.namelock(user, expires, toID(namelock), false, `Autonamelock: ${name}: ${reason}`);
```
What are the parentheses for?

Reply:
They're necessary for destructuring assignment like this: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

Reply:
Oh, I didn't realize it was different if you had already declared the variable.

Reply:
Yeah, surprised me too, never used this pattern before